packages/cli/generators/controller/index.js

```diff
@@ -34,25 +65,150 @@ module.exports = class ControllerGenerator extends ArtifactGenerator {
     return super.promptArtifactName();
   }
 
+  promptArtifactType() {
+    debug('Prompting for controller type');
+    return this.prompt([
+      {
+        type: 'list',
+        name: 'controllerType',
+        message: 'What kind of controller would you like to generate?',
+        when: this.artifactInfo.controllerType === undefined,
+        choices: [ControllerGenerator.BASIC, ControllerGenerator.CRUD],
+        default: ControllerGenerator.BASIC,
+      },
+    ])
+      .then(props => {
+        Object.assign(this.artifactInfo, props);
+        return props;
+      })
+      .catch(err => {
+        debug(`Error during controller type prompt: ${err.stack}`);
+        return this.exit(err);
+      });
+  }
+
+  promptArtifactCrudVars() {
+    let modelList = [];
+    let repositoryList = [];
+    if (
+      !this.artifactInfo.controllerType ||
+      this.artifactInfo.controllerType === ControllerGenerator.BASIC
+    ) {
+      return;
+    }
+    return utils
+      .getArtifactList(this.artifactInfo.modelDir, 'model')
+      .then(list => {
+        if (_.isEmpty(list)) {
+          return this.exit(
+            new Error(`No models found in ${this.artifactInfo.modelDir}`)
+          );
+        }
+        modelList = list;
+        return utils.getArtifactList(
+          this.artifactInfo.repositoryDir,
+          'repository',
+          true
+        );
+      })
+      .then(list => {
+        if (_.isEmpty(list)) {
+          return this.exit(
+            new Error(
+              `No repositories found in ${this.artifactInfo.repositoryDir}`
+            )
+          );
+        }
+        repositoryList = list;
+        return this.prompt([
+          {
+            type: 'list',
+            name: 'modelName',
+            message:
+              'What is the name of the model to use with this CRUD repository?',
+            choices: modelList,
+            when: this.artifactInfo.modelName === undefined,
+            default: modelList[0],
+            validate: utils.validateClassName,
+          },
+          {
+            type: 'list',
+            name: 'repositoryName',
+            message: 'What is the name of your CRUD repository?',
+            choices: repositoryList,
+            when: this.artifactInfo.repositoryName === undefined,
+            default: repositoryList[0],
+            validate: utils.validateClassName,
+          },
+          {
+            type: 'list',
+            name: 'id',
+            message: 'What is the type of your ID?',
+            choices: ['number', 'string', 'object'],
```

Comment:
Ideally, we should be able to the type of model's `id` property automatically. Just saying, an explicit prompt is fine with me for MVP.

Reply:
We might need to build a separate module for the code that constructs the runtime type metadata so that we can share it across the juggler/relations, the openapi-spec and the CLI (since they're all going to need to know this information for some purpose).